components/automate-deployment/pkg/client/restore.go

```diff
@@ -196,65 +198,69 @@ func (r *DeploymentRestore) stopDeploymentService() error {
 		)
 	}
 
-	startTime := time.Now()
-	timeout := 10 * time.Second
 	sleepTime := 500 * time.Millisecond
-	ctx := context.Background()
+	doneChan := make(chan struct{})
 
 	// Wait for Habitat to "unload" the process
-	for {
-		deployed, err := r.target.DeployedServices(ctx)
-		if err == nil {
-			alive := false
-			for _, svc := range deployed {
-				if svc.Pkg.Name() == deploymentServiceName {
-					alive = true
+	go func() {
+		for {
+			deployed, err := r.target.DeployedServices(ctx)
+			if err == nil {
```

Comment:
This is existing code, but I wonder if it might be a bit easier to read if we did

```golang
if err != nil {
    // Maybe log?
    // Maybe sleep  a sec?
    continue
}
```

and then outdent the rest of the code.